pkg/lightning/backend/tidb/tidb.go

```diff
@@ -753,8 +821,33 @@ stmtLoop:
 			err    error
 		)
 		for i := 0; i < writeRowsMaxRetryTimes; i++ {
-			stmt := stmtTask.stmt
-			result, err = be.db.ExecContext(ctx, stmt)
+			query := stmtTask.stmt
+			if be.stmtCache != nil {
+				var prepStmt *sql.Stmt
+				key := &stmtKey{query: query}
+				be.stmtCacheMutex.RLock()
+				stmt, ok := be.stmtCache.Get(key)
+				be.stmtCacheMutex.RUnlock()
+				if ok {
+					prepStmt = stmt.(*sql.Stmt)
+				} else if stmt, err := be.db.PrepareContext(ctx, query); err == nil {
+					prepStmt = stmt
+					be.stmtCacheMutex.Lock()
+					// check again if the key is already in the cache
+					// to avoid override existing stmt without closing it
+					if _, ok := be.stmtCache.Get(key); !ok {
+						be.stmtCache.Put(key, stmt)
+					} else {
+						defer prepStmt.Close()
```

Comment:
we can let `prepStmt` be assigned to the cached entry, and directly Close `stmt` here.